caom2repo/caom2repo/core.py

```diff
@@ -602,14 +602,24 @@ def multiprocess_observation_id(collection, observationID, plugin, subject,
                 "{} - To fix the problem, please add the **kwargs "
                 "argument to the list of arguments for the update"
                 " method of your plugin.".format(str(e)))
+        else:
+            # other unexpected TypeError
+            raise e
     except Exception as e:
-        failed = observation.observation_id
-        rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
+        if observation is None:
+            rootLogger.error('Failed to get observation using observationID = {} - Reason: {}'.format(observationID, e))
+            raise e
```

Comment:
Maybe it should raise e only with halt on error? And move outside the if/case at the end.